relay-pii/src/attachments.rs

```diff
@@ -513,6 +513,27 @@ impl<'a> PiiAttachmentsProcessor<'a> {
             false
         }
     }
+
+    /// Applies PII rules to the given JSON.
+    ///
+    /// This function will perform PII scrubbing using `serde_transcode`, which means that it
+    /// does not have to read the entire document in memory but will rather perform in on a
+    /// per-item basis using a streaming approach.
+    ///
+    /// Returns a scrubbed copy of the JSON document.
+    pub fn scrub_json(&self, payload: &[u8]) -> Result<Vec<u8>, JsonScrubError> {
+        let output = Vec::new();
+
+        let visitor = JsonScrubVisitor::new(self.compiled_config);
+
+        let mut deserializer_inner = Deserializer::from_slice(payload);
+        let deserializer = crate::transform::Deserializer::new(&mut deserializer_inner, visitor);
```

Comment:
nit, to make it super explicit which deserializer is which:

```suggestion
        let mut deserializer_inner = serde_json::Deserializer::from_slice(payload);
        let deserializer = transform::Deserializer::new(&mut deserializer_inner, visitor);
```

Reply:
Good idea!